09_timer/timer.rb

```diff
@@ -0,0 +1,28 @@
+class Timer
+  attr_writer :seconds
+  attr_writer :time_string
+
+  def seconds
+    @seconds = 0
+  end
+
+  def time_string()
+    case @seconds
+    when 0
+      @time_string = '00:00:00'
+    when 0..59
+      @time_string = '00:00:' + padded(@seconds)
+    when 60..3599
+      @time_string = '00:' + padded(@seconds / 60) + ':' + padded(@seconds % 60)
+    else
+      @time_string = padded(@seconds / 3600) + ':' + padded((@seconds - 3600 * (@seconds / 3600)) / 60) + ':' + padded((@seconds - 3600 * ((@seconds / 3600))) % 60)
+    end
```

Comment:
```
"#{pad(hours)}:#{pad(minutes)}:#{pad(seconds)}"
```

Where hours / minutes / seconds are other methods or variables that are calculated based on the seconds instance variable.

Reply:
Extract numbers and give them a name (these are called "magic numbers")

https://refactoring.guru/replace-magic-number-with-symbolic-constant